wallet/src/actors/app/handlers/create_data_req.rs

```diff
@@ -3,21 +3,107 @@ use serde::{Deserialize, Serialize};
 
 use crate::actors::app;
 use crate::types;
+use crate::types::{Hashable as _, ProtobufConvert as _};
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct CreateDataReqRequest {
-    pub rad_request: types::RADRequest,
+    session_id: types::SessionId,
+    wallet_id: String,
+    label: Option<String>,
+    request: DataRequestOutput,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct DataRequestOutput {
+    data_request: RADRequest,
+    value: u64,
+    witnesses: u16,
+    backup_witnesses: u16,
+    commit_fee: u64,
+    reveal_fee: u64,
+    tally_fee: u64,
+    time_lock: u64,
```

Comment:
Can you move this field to inside `RadRequest`? Mainly for consistency with the node.